tests/unit/importer/test_workbook_importer.py

```diff
@@ -0,0 +1,307 @@
+        project_content = list(project_map.values())[0].get('content')
+        self.assertEqual('test project', project_content.get('description'))
+        self.assertEqual(errors, [])
+
+        # and:
+        contributors = project_content.get('contributors')
+        self.assertEqual(2, len(contributors))
+        self.assertIn({'name': 'John', 'email': 'jsmith@email.com'}, contributors)
+        self.assertIn({'name': 'Peter', 'email': 'peterpan@email.com'}, contributors)
+
+    @patch('ingest.importer.importer.WorksheetImporter')
+    def test_do_import_multiple_projects(self, worksheet_importer_constructor):
+        # given:
+        template_mgr = MagicMock(name='template_manager')
+        template_mgr.template.json_schemas = self.mock_json_schemas
+        template_mgr.get_concrete_type = MagicMock(return_value='project')
+
+        worksheet_importer = WorksheetImporter(template_mgr)
+        worksheet_importer_constructor.return_value = worksheet_importer
+        no_errors = []
+        expected_error = {
+            'location': 'sheet=Project',
+            'type': 'MultipleProjectsFound',
+            'detail': 'The spreadsheet should only be associated to a single project.'
+        }
+
+        # and:
+        project_1 = MetadataEntity(concrete_type='project', domain_type='project', object_id=1)
+        project_2 = MetadataEntity(concrete_type='project', domain_type='project', object_id=2)
+        worksheet_importer.do_import = MagicMock(side_effect=[([project_1, project_2], no_errors)])
+
+        # and:
+        workbook = create_test_workbook('Project')
+        workbook_importer = WorkbookImporter(template_mgr)
+
+        # when:
+        is_update = False
+        spreadsheet_json, errors = workbook_importer.do_import(IngestWorkbook(workbook), is_update)
+
+        # then:
+        self.assertIn(expected_error, errors, f'Errors expected to contain {MultipleProjectsFound.__name__}.')
+
+    @patch('ingest.importer.importer.WorksheetImporter')
+    def test_do_import_no_projects(self, worksheet_importer_constructor):
+        # given:
+        template_mgr = MagicMock(name='template_manager')
+        worksheet_importer = WorksheetImporter(template_mgr)
+        worksheet_importer_constructor.return_value = worksheet_importer
+        no_errors = []
+        expected_error = {
+            'location': 'File',
+            'type': 'NoProjectFound',
+            'detail': 'The spreadsheet should be associated to a project.'
+        }
+
+        # and:
+        item = MetadataEntity(concrete_type='product', domain_type='product', object_id=910)
+        worksheet_importer.do_import = MagicMock(side_effect=[([item], no_errors)])
+
+        # and:
+        workbook = create_test_workbook('Item')
+        workbook_importer = WorkbookImporter(template_mgr)
+
+        # when:
+        is_update = False
+        spreadsheet_json, errors = workbook_importer.do_import(IngestWorkbook(workbook), is_update)
```

Comment:
Not sure I understand what is tested here. I see a mock of do_import and then an assertion on its return value. What is the idea behind this test?